docs/examples/external_input.ipynb

```diff
@@ -0,0 +1,209 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# ExternalSource operator\n",
+    "\n",
+    "In this example, we will see how to use `ExternalSource` operator, that allows us to\n",
+    "use an external data source as an input to the Pipeline.\n",
+    "\n",
+    "In order to achieve that, we have to define a Iterator or Generator class which `next` function will\n",
+    "return one or several `numpy` arrays."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import types\n",
+    "import collections\n",
+    "import numpy as np\n",
+    "from random import shuffle\n",
+    "from nvidia.dali.pipeline import Pipeline\n",
+    "import nvidia.dali.ops as ops            \n",
+    "import nvidia.dali.types as types\n",
+    "\n",
+    "batch_size = 16"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Defining the iterator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ExternalInputIterator(object):\n",
+    "    def __init__(self, batch_size):\n",
+    "        self.images_dir = \"images/\"\n",
+    "        self.batch_size = batch_size\n",
+    "        with open(self.images_dir + \"file_list.txt\", 'r') as f:\n",
+    "            self.files = [line.rstrip() for line in f if line is not '']\n",
+    "        shuffle(self.files)\n",
+    "\n",
+    "    def __iter__(self):\n",
+    "        self.i = 0\n",
+    "        self.n = len(self.files)\n",
+    "        return self\n",
+    "\n",
+    "    def __next__(self):\n",
```

Comment:
It would be good to show how we can provide labels corresponding to data loaded here

Reply:
Done